tools/depends/target/date/Makefile

```diff
@@ -0,0 +1,37 @@
+include ../../Makefile.include
+DEPS= ../../Makefile.include Makefile
+
+LIBNAME=date
+VERSION=2.4.1
+ARCHIVE=v$(VERSION).tar.gz
+BASE_URL := https://github.com/HowardHinnant/$(LIBNAME)/archive
```

Comment:
So ive found that platforms that would look to use this will fail to build.

```
/macos/xbmc/XBDateTime.cpp:619:21: error: no member named 'c_encoding' in 'date::weekday'
  time.tm_wday = wd.c_encoding();
                 ~~ ^
```

c_encoding was introduced with the following commit https://github.com/HowardHinnant/date/commit/4e7e76b98125e162e5f0b9e3961668b635122b56
Which is only available in v3.0.0 tag or newer

Reply:
Here is another way to spell `wd.c_encoding()`:
```c++
(wd - Sunday)/days{1}
```

Reply:
I think we have zero benefits to stick with 2.4.1. Let's follow the latest tagged version policy as Howard recommends in the docs.

Reply:
The cmake external project and the depends build should use the same version.